src/main/java/internet/shop/service/impl/OrderServiceImpl.java

```diff
@@ -1,8 +1,51 @@
 package internet.shop.service.impl;
 
+import internet.shop.dao.OrderDao;
+import internet.shop.db.Storage;
+import internet.shop.lib.Inject;
 import internet.shop.lib.Service;
+import internet.shop.model.Order;
+import internet.shop.model.ShoppingCart;
 import internet.shop.service.OrderService;
+import internet.shop.service.ShoppingCartService;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order newOrder = new Order(shoppingCart.getUserId());
+        newOrder.setProducts(List.copyOf(shoppingCart.getProducts()));
+        orderDao.create(newOrder);
+        shoppingCartService.clear(shoppingCart);
+        return newOrder;
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return Storage.orders.stream()
+                .filter(order -> order.getUserId().equals(userId))
```

Comment:
be more attentive about where getUserOrders() logic should be implemented 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_14_2.html